src/sentry/relay/config/metric_extraction.py

```diff
@@ -247,20 +224,30 @@ def _trim_if_above_limit(
     project: Project,
     widget_type: str,
 ) -> list[HashedMetricSpec]:
-    """If we have more specs than the max limit we should trim it."""
+    """If any of the spec versions are above the max limit we should trim it."""
```

Comment:
Good catch. I changed it to this:

```
If the specs are above the max limit we should report it but only trim it for the default spec version.
    This is because running multiple versions should not cause any data loss for the customer."""
```